chainstate/storage/src/internal/test.rs

```diff
@@ -112,23 +107,27 @@ fn test_storage_manipulation() {
     );
 
     // Test setting and retrieving best chain id
-    assert_eq!(store.get_best_block_id(), Ok(None));
-    assert_eq!(store.set_best_block_id(&block0.get_id().into()), Ok(()));
-    assert_eq!(store.get_best_block_id(), Ok(Some(block0.get_id().into())));
-    assert_eq!(store.set_best_block_id(&block1.get_id().into()), Ok(()));
-    assert_eq!(store.get_best_block_id(), Ok(Some(block1.get_id().into())));
+    assert_eq!(db_tx.get_best_block_id(), Ok(None));
+    assert_eq!(db_tx.set_best_block_id(&block0.get_id().into()), Ok(()));
+    assert_eq!(db_tx.get_best_block_id(), Ok(Some(block0.get_id().into())));
+    assert_eq!(db_tx.set_best_block_id(&block1.get_id().into()), Ok(()));
+    assert_eq!(db_tx.get_best_block_id(), Ok(Some(block1.get_id().into())));
 }
 
 #[test]
 fn get_set_transactions() {
```

Comment:
There's an issue that I don't understand yet... this test some times fails, some times succeeds... It needs investigation because I think there's a race. The bad part there is that it it succeeds even when it fails, because the unwrap() that fails is in the thread, and it's not caught.